modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -442,6 +442,8 @@ def where(self, cond, other, **kwargs):
         assert isinstance(
             cond, type(self)
         ), "Must have the same QueryCompiler subclass to perform this operation"
+        # it's doesn't work if `other` is Series._query_compiler because
+        # `n_ary_op` performs columns copartition both for `cond` and `other`.
```

Comment:
```suggestion
        # it's doesn't work if `other` is Series._query_compiler because
        # `n_ary_op` performs columns copartition both for `cond` and `other`.
        if isinstance(other, type(self)) and other.shape_hint is not None:
            other = other.to_pandas()
```